test/core.jl

```diff
@@ -690,6 +690,18 @@ let A = [1]
     @test x == 1
 end
 
+# Pointer finalizer (issue #15408)
+let A = [1]
+    function f(to_finalize::Array{Int64,1})
+      to_finalize[1] = 0
+      nothing
+    end
+    cf = cfunction(f, Void, (Array{Int64,1},))
+    finalizer(A, cf)
```

Comment:
The C finalizer should take a `jl_value_t*` as argument, so I would need to include julia.h from ccalltest.c. Would that be acceptable?